_layouts/default.html

```diff
@@ -9,7 +9,9 @@
 <body>
   <a class="skip-to-main" href="#main-content">Skip to main content</a>
   {% include icons/icons.html %}
-  {% include components/sidebar.html %}
+  {% if site.nav_enabled != false and layout.nav_enabled != false and page.nav_enabled != false %}
```

Comment:
Looking at this boolean condition (and after doing some testing), I'm not sure if this is the right approach. In particular, this doesn't overriding parents that explicitly set `false`:

- if `site.nav_enabled: false`, you cannot override with either `nav_enabled: true` as either a `layout` item or in the `page` front matter
- if `site.nav_enabled: true` but `layout.nav_enabled: false`, you cannot override with `nav_enabled: true` in the page's front matter

In particular, I think this ends up being contradictory with the documentation (which implies that `nav_enabled` can be overridden by layouts/pages). Something that feels more natural to me is

```
page.nav_enabled == true || (page.nav_enabled != false && layout.nav_enabled == true) || (page.nav_enabled != false && layout.nav_enabled != false && site.nav_enabled == true)
```

(this is not valid liquid code, which is a bit of a headache due to the [lack of parens as precedence operators](https://shopify.github.io/liquid/basics/operators/))

What are your thoughts? I may be misinterpreting the intended use of this feature!

---

Here's a different way to interpret this (`F` is anything falsy, so includes `nil`):

| `site.nav_enabled` | `layout.nav_enabled` | `page.nav_enabled` | current PR behaviour | proposed behaviour |
| --- | --- | --- | --- | --- |
| `F` | `F` | `F` | `F` | `F` |
| `F` | `F` | `T` | `F` | `T` |
| `F` | `T` | `F` | `F` | `F` |
| `F` | `T` | `T` | `F` | `T` |
| `T` | `F` | `F` | `F` | `F` |
| `T` | `F` | `T` | `F` | `F` |
| `T` | `T` | `F` | `F` | `F` |
| `T` | `T` | `T` | `T` | `T` |

Reply:
> Something that feels more natural to me is
> 
> ```
> page.nav_enabled == true || (page.nav_enabled != false && layout.nav_enabled == true) || (page.nav_enabled != false && layout.nav_enabled != false && site.nav_enabled == true)
> ```
> 
> (this is not valid liquid code, which is a bit of a headache due to the [lack of parens as precedence operators](https://shopify.github.io/liquid/basics/operators/))

Liquid's `if ... elsif ... endif` corresponds to the usual interpretation of `... || ...`, and circumvents the need for parentheses.

> What are your thoughts? I may be misinterpreting the intended use of this feature!
>
> Here's a different way to interpret this (`F` is anything falsy, so includes `nil`):
>
> | `site.nav_enabled` | `layout.nav_enabled` | `page.nav_enabled` | current PR behaviour | proposed behaviour |
> | - | - | - | - | - |
> | `F` | `F` | `F` | `F` | `F` |

When the variables _all_ have value `nil` (e.g., by not having been set) the intention is for the sidebar to appear as normal, which is inconsistent with the above behaviour.

When `site.nav_enabled` is set to `false`, and the other variables are `false` or `nil`, the sidebar should indeed not appear. So the distinction between `nil` and `false` is significant.

I suggest the following replacement for the conditional inclusion of the sidebar:

```liquid
  {% if page.nav_enabled == true %}
    {% include components/sidebar.html %}
  {% elsif layout.nav_enabled == true and page.nav_enabled == nil %}
    {% include components/sidebar.html %}
  {% elsif site.nav_enabled != false and layout.nav_enabled == nil and page.nav_enabled == nil %}
    {% include components/sidebar.html %}
  {% endif %}
```

Testing `X != false`  instead of `X == nil` above  would give the same behaviour (assuming that the value of `X` is `true`, `false`, or `nil`) but the latter seems clearer.

Reply:
### Testing

Update `docs/layout/minimal/minimal-child.md` to:

```yaml
layout: default
nav_enabled: false
```

Update `docs/layout/minimal/default-child.md` to:

```yaml
layout: minimal
nav_enabled: true
```

1. Check that overriding works as intended for individual pages: the appearance of all three pages in `docs/layout/minimal/` should be as in the v0.8.1 docs, regardless of any setting of `nav_enabled` in `_config.yml`.
2. Check that setting `nav_enabled` in `_config.yml` works as intended: when unset or set to `true`, the sidebar _should_ appear on other pages with `layout: default`; when set to `false`, the sidebar should _not_ appear.

Reply:
Oh, I hadn't thought through all the logical implications in my last changeset. Good catch!

For @mattxwang your bullet point:

> if `site.nav_enabled: true` but `layout.nav_enabled: false`, you cannot override with `nav_enabled: true` in the page's front matter

Thinking aloud, I believe the original reasoning for not displaying the sidebar for this combination is to ensure a consistent display for pages of a given layout.  But I think the behavior is more predictable if we treat `page.nav_enabled` as the most important variable, then `layout.nav_enabled` as slightly less important, and `site.nav_enabled` as least important.

So this line in your truth table would have the proposed behavior be `T` to display the sidebar.

| `site.nav_enabled` | `layout.nav_enabled` | `page.nav_enabled` | current PR behaviour | proposed behaviour |
|-|-|-|-|-|
| `T` | `F` | `T` | `F` | `T` |

@pdmosses and your ideal (but not liquid-feasible) code both seem to agree with this proposed behavior but I just wanted to make this part of the specification explicit before pushing a change.

Does this sound good?